### DISCUSSION TITLE
How to use responsive style props for `size` attribute?

### DISCUSSION THREAD

**Discussion by basememara** [mantinedev/mantine#7137] [Q&A]
I keep wanting to use the cool [response style props](https://mantine.dev/styles/responsive/) for the `size` attribute but haven't found a good alternative. I'm trying to do something like this:

```react
<Stepper size={{ base: 'xs', sm: 'md', lg: 'xl' }} />
```

I tried using `useMediaQuery` and the `sx`, but they are verbose and sometimes needed in SSR components. Is there a reason why this the responsive style props aren't available for `size` or am I missing a more idiomatic way to achieve this?

Thanks for an amazing and beautiful library!

**Comment by rtivital** [comment]
https://mantine.dev/styles/responsive/#component-size-based-on-media-query
